pkg/signatures/cosign_sig_verifier.go

```diff
@@ -201,13 +228,16 @@ func (c *cosignSignatureVerifier) createVerifierOpts() error {
 			verifierErrs = multierror.Append(verifierErrs, errors.Wrap(err, "creating verifier"))
 			continue
 		}
-		opts := defaultCosignCheckOpts()
+		opts, err := c.defaultCosignCheckOpts(ctx)
+		if err != nil {
+			return err
+		}
 		opts.SigVerifier = v
 		c.verifierOpts = append(c.verifierOpts, opts)
 	}
 
 	for _, certs := range c.certs {
-		opts, err := cosignCheckOptsFromCert(certs)
+		opts, err := c.cosignCheckOptsFromCert(ctx, certs)
```

Comment:
This could probably be refactored:
* the method `cosignCheckOptsFromCert` receives the receiver's `c` property `c.certs[i]` as an argument;
* the first check in the method does not use `certs`, so doesn't need to be checked in a loop.